src/libsystemd-network/ndisc-option.c

```diff
@@ -1358,6 +1358,8 @@ static int ndisc_option_parse_encrypted_dns(Set **options, size_t offset, size_t
         r = ndisc_get_dns_name(opt + off, ilen, &res.auth_name);
         if (r < 0)
                 return r;
+        if (!in_charset(res.auth_name, URI_UNRESERVED))
+                return -EBADMSG;
```

Comment:
It's probably unrelated to this PR but I ran into embedded dot in https://github.com/systemd/systemd/pull/30952#discussion_r1577838543. `networkctl` and `resolvectl` show things like `2001:db8::1234#001000003..b039` while it's `2001:db8::1234#\001\000\003.\.b\039` in `/run/systemd/netif/state`. (I'm trying to figure out what's up with ENODATA though so the embedded dots are on the backburner)

Reply:
Hm yeah embedded dot is kind of a pain.